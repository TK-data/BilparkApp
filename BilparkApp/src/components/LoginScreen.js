import React from 'react';
import PropTypes from 'prop-types';
import { Button, StyleSheet, Text, TextInput, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});

// const xLoginScreen = ({ navigation }) => (
//   <View style={styles.container}>
//     <Text style={styles.welcome}>
//       This page posts data to server for login
//     </Text>
//     <TextInput
//     />
//     <Button
//       onPress={() => navigation.dispatch({ type: 'Login' })}
//       title="Log in"
//     />
//   </View>
// );

class LoginScreen extends React.Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
  };

  static navigationOptions = {
    title: 'Log in',
  };

  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
    };

  }

  login = () => {
    const { email, password } = this.state;

    // this.props.actions.loginRequest(email, password);


    // the loginRequest (or userService?) will handle the request and
    // notify the navigation about change if login is success?
  }

  render() {
    // const navigation = this.props.navigation;

    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          This page posts data to server for login
        </Text>
        <Text value={this.state.email} />
        <TextInput
          value={this.state.email}
          onChangeText={email => this.setState({ email })}
        />
        <TextInput
          value={this.state.password}
          onChangeText={password => this.setState({ password })}
          secureTextEntry
        />
        <Button
          /* onPress={() => navigation.dispatch({ type: 'Login' })} */
          onPress={this.login}
          title="Log in"
        />
      </View>
    );
  }
}

// xLoginScreen.propTypes = {
//   navigation: PropTypes.object.isRequired,
// };
//
// xLoginScreen.navigationOptions = {
//   title: 'Log In',
// };

export default LoginScreen;
