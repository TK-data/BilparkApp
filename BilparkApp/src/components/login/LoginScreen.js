import React from 'react';
import { Body, Content } from 'native-base';
import { StyleSheet, View, Image } from 'react-native';
import { connect } from 'react-redux';

import LoginForm from './LoginForm';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#002776',
  },
  content: {
    flex: 3,
    justifyContent: 'flex-start',
    alignItems: 'center',
    backgroundColor: '#002776',
  },
  button: {
    alignSelf: 'center',
  },
  contentImage: {
    flex: 1,
    justifyContent: 'center',
  },
  contentContainer: {
    flex: 2,
    justifyContent: 'center',
  },
  carlogo: {
    height: 150,
    width: 150,
  },
});

class LoginScreen extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.contentImage}>
          <Image resizeMode="contain" style={styles.carlogo} source={require('../../images/car.png')} />
        </View>
        <View style={styles.contentContainer}>
          <Content contentContainerStyle={styles.content}>
            <LoginForm />
          </Content>
        </View>
      </View>
    );
  }
}


const mapStateToProps = () => {
  return {
  };
};

const mapDispatchToProps = () => ({
});


LoginScreen.navigationOptions = {
  title: 'Logg inn',
  headerLeft: <Body />,
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginScreen);
