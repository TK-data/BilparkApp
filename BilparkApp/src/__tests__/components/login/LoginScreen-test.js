import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import LoginScreen from '../../../components/login/LoginScreen';

const initialState = {
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing LoginScreen', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <LoginScreen />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });
});
