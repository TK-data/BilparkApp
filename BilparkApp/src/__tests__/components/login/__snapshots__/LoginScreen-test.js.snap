// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginScreen Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "height": 150,
                "width": 150,
              }
            }
          />
        </View>,
        <View
          style={
            Object {
              "flex": 2,
              "justifyContent": "center",
            }
          }
        >
          <Styled(Content)
            contentContainerStyle={
              Object {
                "alignItems": "center",
                "backgroundColor": "#002776",
                "flex": 3,
                "justifyContent": "flex-start",
              }
            }
            virtual={undefined}
          >
            <Connect(LoginForm) />
          </Styled(Content)>
        </View>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#002776",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Image
            resizeMode="contain"
            source={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "height": 150,
                "width": 150,
              }
            }
          />,
          "style": Object {
            "flex": 1,
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "resizeMode": "contain",
            "source": Object {
              "process": [Function],
            },
            "style": Object {
              "height": 150,
              "width": 150,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Content)
            contentContainerStyle={
              Object {
                "alignItems": "center",
                "backgroundColor": "#002776",
                "flex": 3,
                "justifyContent": "flex-start",
              }
            }
            virtual={undefined}
          >
            <Connect(LoginForm) />
          </Styled(Content)>,
          "style": Object {
            "flex": 2,
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(LoginForm) />,
            "contentContainerStyle": Object {
              "alignItems": "center",
              "backgroundColor": "#002776",
              "flex": 3,
              "justifyContent": "flex-start",
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 150,
                  "width": 150,
                }
              }
            />
          </View>,
          <View
            style={
              Object {
                "flex": 2,
                "justifyContent": "center",
              }
            }
          >
            <Styled(Content)
              contentContainerStyle={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#002776",
                  "flex": 3,
                  "justifyContent": "flex-start",
                }
              }
              virtual={undefined}
            >
              <Connect(LoginForm) />
            </Styled(Content)>
          </View>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#002776",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              resizeMode="contain"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 150,
                  "width": 150,
                }
              }
            />,
            "style": Object {
              "flex": 1,
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "resizeMode": "contain",
              "source": Object {
                "process": [Function],
              },
              "style": Object {
                "height": 150,
                "width": 150,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Content)
              contentContainerStyle={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#002776",
                  "flex": 3,
                  "justifyContent": "flex-start",
                }
              }
              virtual={undefined}
            >
              <Connect(LoginForm) />
            </Styled(Content)>,
            "style": Object {
              "flex": 2,
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(LoginForm) />,
              "contentContainerStyle": Object {
                "alignItems": "center",
                "backgroundColor": "#002776",
                "flex": 3,
                "justifyContent": "flex-start",
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
